__tests__/components/RentalFee.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * (C) Symbol Contributors 2022
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+import RentalFee from '@/components/RentalFees/RentalFee.vue';
+import { render } from '@testing-library/vue';
+import { getStore } from '@MOCKS/Store';
+import i18n from '@/language/index';
+import { mosaicsMock } from '@MOCKS/mosaics';
+import { UInt64 } from 'symbol-sdk';
+
+describe('components/RentalFee', () => {
+    const renderComponent = (stateChanges = {}, props = {}) => {
+        const mockNetworkStore = {
+            namespaced: true,
+            state: { rentalFeeEstimation: undefined },
+            getters: {
+                rentalFeeEstimation: (state) => {
+                    return state.rentalFeeEstimation;
+                },
+            },
+        };
+
+        const mockMosaicStore = {
+            namespaced: true,
+            state: {
+                networkCurrency: mosaicsMock[0],
+            },
+            getters: {
+                networkCurrency: (state) => state.networkCurrency,
+            },
+        };
+
+        const store = getStore(
+            {
+                network: mockNetworkStore,
+                mosaic: mockMosaicStore,
+            },
+            stateChanges,
+            undefined,
+            true,
+            undefined,
+            false,
+        );
+
+        return {
+            ...render(RentalFee, {
+                store,
+                i18n,
+                props,
+            }),
+            store,
+        };
+    };
+
+    const createRentalFeeEstimation = {
+        rentalFeeEstimation: {
+            effectiveRootNamespaceRentalFeePerBlock: UInt64.fromUint(200),
+            effectiveChildNamespaceRentalFee: UInt64.fromUint(10000000),
+            effectiveMosaicRentalFee: UInt64.fromUint(50000000),
+        },
+    };
+
+    const runBasicEstimatedRentalFee = (stateChanges, rentalType, expectedResult) => {
+        // Arrange:
+        const { getByText, getByTestId } = renderComponent(stateChanges, {
+            rentalType,
+            duration: 100000,
+        });
+
+        // Act + Assert:
+        expect(getByText('Estimated rental fee:')).toBeDefined();
+        expect(getByTestId('fees').textContent.trim()).toContain(expectedResult);
+    };
+
+    it('renders default estimated rental fee when rentalFeeEstimation does not exist', () => {
+        runBasicEstimatedRentalFee({}, 'mosaic', '0');
```

Comment:
why are you checking the rendered text vs the result of rentalFeeAmount ?

Reply:
Recently we had a discussion in our daily meeting, about using `@testing-library/vue`
The test should focus more on user behavior or experience in the wallet.

so **rendered text** will be more accurate because it will display to the user.

What do you think about it?

Reply:
can you please summarize this discussion / decision in a discord channel for further discussion?